### DISCUSSION TITLE
which CUDA APIs have been used during mining

### DISCUSSION THREAD

**Discussion by ALEX7-XX** [filecoin-project/community#121] [Q&A]
Dear Mr/Miss:

I’m trying to find which CUDA APIs have been used during mining?

Cause P1 S1 S2 stages can be accelerated by GPU, Nvidia Card, I want to figure out how does lotus use GPU to accelerate the mining?

Best Regard

**Comment by rjan90** [comment]
It is probably better to ask in, or check out [https://github.com/filecoin-project/rust-fil-proofs](https://github.com/filecoin-project/rust-fil-proofs) for more information about the GPU-code related to mining.
